File: go/tests/test_ipv6_loopback.go

import "math/big"

import "./ipaddress"

    type IPv6LoopbackTest struct {
        ip IPAddress,
        s string,
        n string,
        _str string,
        one: big.Int,
    }

    func setup() IPv6LoopbackTest {
        return IPv6LoopbackTest {
            ipv6_loopback::new(),
            "::1",
            "::1/128",
            "0000:0000:0000:0000:0000:0000:0000:0001/128",
            big.Int.new(1),
        };
    }

int main() {
  describe("test_ipv6_loopback", func() {
    it("test_attributes", func() {
        let s = setup();
        assert_eq!(128, s.ip.prefix.num);
        assert_eq!(true, s.ip.is_loopback());
        assert_eq!(s.s, s.ip.to_s());
        assert_eq!(s.n, s.ip.to_string());
        assert_eq!(s.string, s.ip.to_string_uncompressed());
        assert_eq!(s.one, s.ip.host_address);
    })

    it("test_method_ipv6", func() {
        assert_eq!(true, setup().ip.is_ipv6());
    })
  });
}

File: python/tests/test_ipv6_loopback.py
from ip_address.ipv6_loopback import Ipv6Loopback


class IPv6LoopbackTest:
    def __init__(self, ip, s, n, string, one):
        self.ip = ip
        self.s = s
        self.n = n
        self.string = string
        self.one = one


def setup():
    return IPv6LoopbackTest(
        ip=Ipv6Loopback.create(),
        s="::1",
        n="::1/128",
        string="0000:0000:0000:0000:0000:0000:0000:0001/128",
        one=1,
    )


def test_attributes():
    s = setup()
    assert 128 == s.ip.prefix.num
    assert s.ip.is_loopback() is True
    assert s.s == s.ip.to_s()
    assert s.n == s.ip.to_string()
    assert s.string == s.ip.to_string_uncompressed()
    assert str(s.one) == str(s.ip.host_address)


def test_method_ipv6():
    assert setup().ip.is_ipv6() is True
